### PR TITLE
[Engine][Task] Use TaskGroupLocation instead of TaskGroupID

### PR DESCRIPTION
**Pull request by ic4y**
Use TaskGroupInfo instead of TaskGroupID.
Solve the problem of the same TaskGroupId of different Jobs with different PipelineIds.
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by EricJoy2048** [comment]
LGTM

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/sink/SinkRegisterOperation.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public void run() throws Exception {
         SinkAggregatedCommitterTask<?> task = null;
         for (int i = 0; i < RETRY_TIME; i++) {
             try {
-                task = server.getTaskExecutionService().getExecutionContext(committerTaskID.getTaskGroupID())
```
Comment (ashulin) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/sink/SinkRegisterOperation.java:59]: Can the encapsulation of the group-id be improved? Hide details in TaskExecutionService, TaskID is unique for a job, task-group-id & pipeline-id can be hidden.

Comment (ic4y) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/sink/SinkRegisterOperation.java:59]: Our current TaskID is not unique and may be repeated in different jobs or pipelines

Comment (ashulin) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/sink/SinkRegisterOperation.java:59]: I will create an issue for this as an improvement

**Pull request merged**